pkg/cloudprovider/vsphere/nodemanager.go

```diff
@@ -494,14 +507,19 @@ func findSubnetMatch(ipAddrNetworkNames []*ipAddrNetworkName, networkSubnet *net
 // given network name, ignoring case.
 func findNetworkNameMatch(ipAddrNetworkNames []*ipAddrNetworkName, networkName string) *ipAddrNetworkName {
 	if networkName != "" {
-		networkNameMatches := filter(ipAddrNetworkNames, func(candidate *ipAddrNetworkName) bool {
+		return findFirst(ipAddrNetworkNames, func(candidate *ipAddrNetworkName) bool {
 			return strings.EqualFold(networkName, candidate.networkName)
 		})
+	}
+	return nil
+}
 
-		if len(networkNameMatches) > 0 {
-			return networkNameMatches[0]
+// findFrist returns the first occurance that matches the given predicate
+func findFirst(ipAddrNetworkNames []*ipAddrNetworkName, predicate func(*ipAddrNetworkName) bool) *ipAddrNetworkName {
```

Comment:
I see this function replaces the `filter` function, but the filter function still exists. Is there somewhere it is still being used or can it be deleted?

Reply:
filter is still used in 2 places